SOMotionDetector/SOLocationManager.m

```diff
@@ -18,6 +18,9 @@ - (id)init
         self.locationManager = [[CLLocationManager alloc] init];
         self.locationManager.desiredAccuracy = kCLLocationAccuracyBest;
         self.locationManager.distanceFilter = kCLDistanceFilterNone;
+        [self.locationManager requestWhenInUseAuthorization];
+        [self.locationManager startMonitoringSignificantLocationChanges];
```

Comment:
You should either use startMonitoringSignificantLocationChanges or startUpdatingLocation but not both in my opinion.